test/enum_if.cpp

```diff
@@ -24,34 +24,34 @@ std::string operator "" _s(char const* str, size_t len) { return std::string(str
 std::string print_flags(aux::interface_flags const f)
 {
 	return
-		((f & if_flags::up) ? "UP "_s : ""_s)
-		+ ((f & if_flags::broadcast) ? "BROADCAST "_s : ""_s)
-		+ ((f & if_flags::loopback) ? "LOOP "_s : ""_s)
-		+ ((f & if_flags::pointopoint) ? "PPP "_s : ""_s)
-		+ ((f & if_flags::running) ? "RUN "_s : ""_s)
-		+ ((f & if_flags::noarp) ? "NOARP "_s : ""_s)
-		+ ((f & if_flags::promisc) ? "PROMISC "_s : ""_s)
-		+ ((f & if_flags::allmulti) ? "ALLMULTI "_s : ""_s)
-		+ ((f & if_flags::master) ? "MASTER "_s : ""_s)
-		+ ((f & if_flags::slave) ? "SLAVE "_s : ""_s)
-		+ ((f & if_flags::multicast) ? "MULTICAST "_s : ""_s)
-		+ ((f & if_flags::dynamic) ? "SYN "_s : ""_s)
-		+ ((f & if_flags::lower_up) ? "LWR_UP "_s : ""_s)
-		+ ((f & if_flags::dormant) ? "DORMANT "_s : ""_s)
+		((f & aux::if_flags::up) ? "UP "_s : ""_s)
+		+ ((f & aux::if_flags::broadcast) ? "BROADCAST "_s : ""_s)
+		+ ((f & aux::if_flags::loopback) ? "LOOP "_s : ""_s)
+		+ ((f & aux::if_flags::pointopoint) ? "PPP "_s : ""_s)
+		+ ((f & aux::if_flags::running) ? "RUN "_s : ""_s)
+		+ ((f & aux::if_flags::noarp) ? "NOARP "_s : ""_s)
+		+ ((f & aux::if_flags::promisc) ? "PROMISC "_s : ""_s)
+		+ ((f & aux::if_flags::allmulti) ? "ALLMULTI "_s : ""_s)
+		+ ((f & aux::if_flags::master) ? "MASTER "_s : ""_s)
+		+ ((f & aux::if_flags::slave) ? "SLAVE "_s : ""_s)
+		+ ((f & aux::if_flags::multicast) ? "MULTICAST "_s : ""_s)
+		+ ((f & aux::if_flags::dynamic) ? "SYN "_s : ""_s)
+		+ ((f & aux::if_flags::lower_up) ? "LWR_UP "_s : ""_s)
+		+ ((f & aux::if_flags::dormant) ? "DORMANT "_s : ""_s)
 		;
 }
 
 char const* print_state(aux::if_state const s)
 {
 	switch (s)
 	{
-		case if_state::up: return "up";
```

Comment:
would you mind sticking this at the top of the function instead?

```
using if_state = aux::if_state;
```

Reply:
(and the function above too)

Reply:
sure, looks better